scrapy_zyte_api/providers.py

```diff
@@ -112,10 +125,36 @@ async def __call__(  # noqa: C901
                     options["extractFrom"] = extract_from.value
                     break
 
+        http_response_needed = (
+            AnyResponse in to_provide
+            and BrowserResponse not in to_provide
+            and BrowserHtml not in to_provide
+            and not http_response
+        )
+
+        extract_from = None  # type: ignore[assignment]
         for item_type, kw in item_keywords.items():
             options_name = f"{kw}Options"
             if item_type not in to_provide_stripped and options_name in zyte_api_meta:
                 del zyte_api_meta[options_name]
+            elif options_name in zyte_api_meta:
+                extract_from = zyte_api_meta[options_name].get("extractFrom")
+            elif item_type in to_provide_stripped and http_response_needed:
+                zyte_api_meta[options_name] = {"extractFrom": "httpResponseBody"}
```

Comment:
To check my understanding: the logic here is that if browserHtml is not requested, but a data type is requested, and there is AnyResponse, then we switch the extractFrom from default to httpResponseBody for this data type?

Reply:
It'd be, if:
- there's no explicit extraction source requested for the given `item_type` (e.g. Product, ProductNavigation, etc), and
- AnyResponse is one of the requested, and
- either BrowserResponse or BrowserHtml is not requested, and
- HttpResponse has not been created by previous providers (i.e. HttpResponseProvider)

then we use `httpResponseBody` as the extraction source.